api/internal/generators/utils.go

```diff
@@ -40,6 +40,9 @@ func makeValidatedDataMap(
 	if err != nil {
 		return nil, errors.WrapPrefix(err, "loading KV pairs", 0)
 	}
+	sort.Slice(pairs, func(i, j int) bool {
+		return pairs[i].Key < pairs[j].Key
```

Comment:
This PR does not attempt too preserve the input order. It is performing natural sorting alphabetically. If I understood correctly both are valid approaches with preference to the other. I believe someone else was working on that. In the meantime this could help solve the issue to provide a predictable behavior. 

If that is unacceptable I can close the PR

Reply:
Sorting alphabetically is fine. But you are putting the values back into a map, and the map does _not_ preserve order; it will return to you the key/value pairs in some arbitrary order, so this sort is not really doing anything.

Reply:
Ok. I now understand your comment better now. But wouldn’t  that fail the unit test? Apologies

Reply:
> Ok. I understand your comment better now. But wouldn’t that fail the unit test? Apologies

The unit test would probably be flaky, but not necessarily always fail or even fail often. For example, when I run the example in the issue in https://github.com/kubernetes-sigs/kustomize/issues/4292#issuecomment-984131055, 9/10 times I get a consistent output. 1/10 times I get something different. 

It depends on go implements the map, we may get lucky and sometimes have the same order each time. But since go does not guarantee the order, I would not rely on it.

Reply:
Got it. Thank you for taking the time to explain. Appreciate it.